### PR TITLE
Add ability to activate CH Table at an arbitrary point

### PR DESCRIPTION
**Pull request by dsouzai**
The CHT is augmented to have a notion of a state. By default, the state is active. However, this PR adds the infrastructure to allow the CH Table to be initially inactive, and activated at a later point in JVM execution. The API to activate the CH Table assumes a stop the world event where the class hierarchy isn't changing. It then calls a helper method which is recursive, as follows:

1. Add superclasses
2. Add interfaces implemented
3. Trigger class load hook
4. Trigger class preinitialize hook
5. Add array class

This code is mostly a port from the snapshot branch.

### CODE REVIEW

**Comment by dsouzai** [comment]
@0xdaryl for your review. @mpirvu fyi.

File: runtime/compiler/env/PersistentCHTable.hpp [hunk 1]
```diff
@@ -46,6 +46,24 @@ namespace TR { class ResolvedMethodSymbol; }
...
    {
+   private:
+   /**
+    * @brief Enum to denote the state of the CH Table:
+    *
+    *        active: Data in the CH Table can be used and updated.
+    *        activating: CH Table is in the process of being activated;
+    *                    this state is used when the CH Table is
+    *                    being populated in the restore run.
+    *        reset: CH Table is not active.
```
Comment (0xdaryl) [inline runtime/compiler/env/PersistentCHTable.hpp:58]: For completeness, add `activationFailed` ?

Comment (0xdaryl) [inline runtime/compiler/env/PersistentCHTable.hpp:58]: Should `reset` be first to match the ordering in the enum?

**Review by 0xdaryl — CHANGES_REQUESTED** [review]
Also, please see the Copyright CI failures.

File: runtime/compiler/env/PersistentCHTable.hpp (hunk 1 above)
Comment (dsouzai) [reply runtime/compiler/env/PersistentCHTable.hpp]: Done, see https://github.com/eclipse/openj9/compare/0efdded6403247d28fca53431cb1a2abe4487fdf..043c3c65ba6a7889d13137fccdf5885a384e3e2e

Comment (dsouzai) [reply runtime/compiler/env/PersistentCHTable.hpp]: Done, see https://github.com/eclipse/openj9/compare/0efdded6403247d28fca53431cb1a2abe4487fdf..043c3c65ba6a7889d13137fccdf5885a384e3e2e

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk11

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**